superset/utils/cache.py

```diff
@@ -14,11 +14,51 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-from typing import Any, Callable, Optional
+import hashlib
+import json
+import logging
+from datetime import datetime
+from typing import Any, Callable, Dict, Optional
 
 from flask import request
+from flask_caching import Cache
 
+from superset import app, utils
 from superset.extensions import cache_manager
+from superset.stats_logger import BaseStatsLogger
+from superset.utils.core import json_int_dttm_ser
+
+config = app.config
+stats_logger: BaseStatsLogger = config["STATS_LOGGER"]
+logger = logging.getLogger(__name__)
+
+# TODO: DRY up cache key code
+def json_dumps(obj: Any, sort_keys: bool = False) -> str:
+    return json.dumps(obj, default=json_int_dttm_ser, sort_keys=sort_keys)
+
+
+def generate_cache_key(values_dict: Dict[str, Any], key_prefix: str = "") -> str:
+    json_data = json_dumps(values_dict, sort_keys=True)
+    hash = hashlib.md5(json_data.encode("utf-8")).hexdigest()
+    return f"{key_prefix}{hash}"
+
+
+def set_and_log_cache(
```

Comment:
@bkyryliuk I'm starting to move/centralize the caching logic, and have not included the `CacheKey` write that is present in the `viz.py` implementation because 1) I don't understand the use case, and 2) we can't afford the performance hit of writing to the metadata DB in the same method that writes to a K/V store (we're currently using Redis). Can you elaborate on `CacheKey` and how it's used?

Reply:
Sure, cache key tables stores the mapping between datasources and cache keys to enable the invalidation by datasource though api endpoint. In our use case we have external ETL systems that are updating tables that are powering the superset charts and we are using this model and API end point to evict / invalidate cache records for the tables that were updated in our ETL systems. It is fairly critical for us to keep data in superset fresh and trustworthy. This endpoint has fairly low traffic as is hit only on the explore hit compares to the logging calls.

Reply:
I would prefer to keep it, it would be fine to hide it behind the feature flag is you expect to experience performance issues

Reply:
I added a `STORE_CACHE_KEYS_IN_METADATA_DB` config value to enable this.